govcd/vm.go

```diff
@@ -107,6 +137,35 @@ func (vm *VM) PowerOn() (Task, error) {
 
 }
 
+// PowerOnAndForceCustomization is a synchronous function which is equivalent to the functionality
+// one has in UI. It triggers customization which may be useful in some cases (like altering NICs)
+//
+// The VM _must_ be Undeployed for this action to actually work.
```

Comment:
Can we detect the state of the VM to make sure this prerequisite is met?

Reply:
IMHO/ Yes, we should. VM should have status.

Reply:
I can add a check, but the problem is that it doesn't actually fail to power on and god knows if it can be useful in any scenario. As far as I've investigated it simply never exits "GC_PENDING" if it wasn't un-deployed before using `PowerOnAndForceCustomization`.

Reply:
Now I lost between explanation and doc:  "The VM _must_ be Undeployed for this action to actually work." ?

Reply:
Ok. On a second read that may not be too clear. Power on works anyway, but the VM never exits `GC_PENDING` state if it was not undeployed.

Reply:
doesn't `vm.status == undeployed` isn't sufficient?

Reply:
It may be. The problem is that if I enforce `vm.status == undeployed`, I'm not 100% sure it is the only case it would be useful.

Reply:
I don't understand. If we can check the status, we can simply abort when the wanted status is not set.
What am I missing?

Reply:
Added.